### PR TITLE
test: cover obvious SASL PLAIN edge cases

### PR DESCRIPTION
**Pull request by dgw**
The major edge cases handled in Sopel's SASL PLAIN code are:

- Unexpected continuation message from the server (`AUTHENTICATE PLAIN` ack'd by something other than `AUTHENTICATE +`)
- Very long password that b64encodes to exactly 400 bytes
- Very long password that b64encodes to over 400 bytes

IRCv3 recently merging [a patch to add an example](https://github.com/ircv3/ircv3-specifications/pull/521) of split token negotiation inspired me to cover that in our behavior tests.

Along the way I fixed a couple typos, and realized it would also be pretty easy to validate Sopel aborting SASL PLAIN on a non-empty message from the server.

### Checklist
- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches
  - This change is literally tests, so it tested itself 😁

### Notes
Big thanks to @Exirel for writing tests of the main SASL PLAIN codepath, which I copied and adapted to cover these edge cases.

### CODE REVIEW

File: test/coretasks/test_coretasks_sasl.py [hunk 1]
```diff
@@ -179,6 +293,33 @@ def test_sasl_plain_not_supported(botfactory: BotFactory, tmpconfig) -> None:
...
+        'AUTHENTICATE PLAIN',
+    )
+
+    # server acknowledges PLAIN auth request in an unexpected way
+    n = len(mockbot.backend.message_sent)
+    mockbot.on_message(':irc.example.com AUTHENTICATE VGVzdEJvdABUZXN0Qm90AG')
+    assert mockbot.backend.message_sent[n:] == rawlist(
+        'AUTHENTICATE *',
+    ), ('Bot must abort SASL PLAIN auth if server reply to starting '
+        'AUTHENTICATE PLAIN is not as expected per SASL spec')
```
Comment (Exirel) [inline test/coretasks/test_coretasks_sasl.py:320]: What is the expected behavior from the server here? Because I'd make sure here we can send CAP END after that cancelation.

**Review by Exirel — COMMENTED** [review]
It looks good to me however I've this nitpick about authenticate cancelation.

File: test/coretasks/test_coretasks_sasl.py (hunk 1 above)
Comment (dgw) [reply test/coretasks/test_coretasks_sasl.py:320]: > The client can abort an authentication by sending an asterisk as the data. The server will send a 906 numeric.

906 (`ERR_SASLABORTED`) is handled by `sasl_fail()`. Sopel will log the failure, then quit with a "SASL Auth Failed" message.

**Review by Exirel — CHANGES_REQUESTED** [review]
Aside from adding the ERR_SASLABORTED to check that we send the CAP END, it's all good.

File: test/coretasks/test_coretasks_sasl.py (hunk 1 above)
Comment (dgw) [reply test/coretasks/test_coretasks_sasl.py:320]: Have done so!

**Review by Exirel — APPROVED** [review]
Perfect, let's go!
